Source/hydro/Castro_mol_nd.F90

```diff
@@ -241,209 +219,101 @@ subroutine ca_mol_single_stage(lo, hi, time, &
-                 eos_state%xn(:)  = qyp(i,j,k,QFS:QFS-1+nspec)
-                 eos_state%aux(:) = qyp(i,j,k,QFX:QFX-1+naux)
-
-                 call eos(eos_input_rt, eos_state)
-
-                 qyp(i,j,k,QPRES) = eos_state%p
-                 qyp(i,j,k,QREINT) = qyp(i,j,k,QRHO)*eos_state%e
-                 ! should we try to do something about Gamma_! on interface?
-
-                 eos_state%rho    = qym(i,j,k,QRHO)
-                 eos_state%T      = qym(i,j,k,QTEMP)
-                 eos_state%xn(:)  = qym(i,j,k,QFS:QFS-1+nspec)
-                 eos_state%aux(:) = qym(i,j,k,QFX:QFX-1+naux)
-
-                 call eos(eos_input_rt, eos_state)
-
-                 qym(i,j,k,QPRES) = eos_state%p
-                 qym(i,j,k,QREINT) = qym(i,j,k,QRHO)*eos_state%e
-                 ! should we try to do something about Gamma_! on interface?
-#endif
-
-#if AMREX_SPACEDIM == 3
-                 eos_state%rho    = qzp(i,j,k,QRHO)
-                 eos_state%T      = qzp(i,j,k,QTEMP)
-                 eos_state%xn(:)  = qzp(i,j,k,QFS:QFS-1+nspec)
-                 eos_state%aux(:) = qzp(i,j,k,QFX:QFX-1+naux)
+                 eos_state%rho    = qp(i,j,k,QRHO,idir)
+                 eos_state%T      = qp(i,j,k,QTEMP,idir)
+                 eos_state%xn(:)  = qp(i,j,k,QFS:QFS-1+nspec,idir)
+                 eos_state%aux(:) = qp(i,j,k,QFX:QFX-1+naux,idir)
 
                  call eos(eos_input_rt, eos_state)
 
-                 qzp(i,j,k,QPRES) = eos_state%p
-                 qzp(i,j,k,QREINT) = qzp(i,j,k,QRHO)*eos_state%e
+                 qp(i,j,k,QPRES,idir) = eos_state%p
+                 qp(i,j,k,QREINT,idir) = qp(i,j,k,QRHO,idir)*eos_state%e
                  ! should we try to do something about Gamma_! on interface?
 
-                 eos_state%rho    = qzm(i,j,k,QRHO)
-                 eos_state%T      = qzm(i,j,k,QTEMP)
-                 eos_state%xn(:)  = qzm(i,j,k,QFS:QFS-1+nspec)
-                 eos_state%aux(:) = qzm(i,j,k,QFX:QFX-1+naux)
+                 eos_state%rho    = qm(i,j,k,QRHO,idir)
+                 eos_state%T      = qm(i,j,k,QTEMP,idir)
+                 eos_state%xn(:)  = qm(i,j,k,QFS:QFS-1+nspec,idir)
+                 eos_state%aux(:) = qm(i,j,k,QFX:QFX-1+naux,idir)
 
                  call eos(eos_input_rt, eos_state)
 
-                 qzm(i,j,k,QPRES) = eos_state%p
-                 qzm(i,j,k,QREINT) = qzm(i,j,k,QRHO)*eos_state%e
+                 qm(i,j,k,QPRES,idir) = eos_state%p
+                 qm(i,j,k,QREINT,idir) = qm(i,j,k,QRHO,idir)*eos_state%e
                  ! should we try to do something about Gamma_! on interface?
-#endif
 
               end do
            end do
         end do
-     endif
+     end do
+  end if
 
-     ! Compute F^x at kc (k3d)
-     call cmpflx(qxm, qxp, It_lo, It_hi, &
-                 flux1, flux1_lo, flux1_hi, &
-                 q1, flux1_lo, flux1_hi, &  ! temporary
+  ! Compute F^x at kc (k3d)
+  call cmpflx(qm(:,:,:,:,1), qp(:,:,:,:,1), It_lo, It_hi, &
```

Comment:
Is it possible to avoid this type of slicing? I recall that OpenACC tends not to like it, and we want to keep that route open.

Reply:
I can either explicitly put a copy here, or I can have the cmpflx routine take the states of this form (with the interface direction as the last dimension) and then select the direction internally.  Preference?

Reply:
I prefer the latter.

Reply:
me too

Reply:
this is fixed now.  `cmpflx` takes two additional arguments, the number of components in the state arrays and the which component we are using

Reply:
tests still pass